planner/core/rule_join_reorder_dp.go

```diff
@@ -286,5 +286,6 @@ func (s *joinReorderDPSolver) newJoinWithConds(leftPlan, rightPlan LogicalPlan,
 	join := s.newCartesianJoin(leftPlan, rightPlan)
 	join.EqualConditions = eqConds
 	join.OtherConditions = otherConds
+	join.cartesianJoin = len(join.EqualConditions) == 0
```

Comment:
IMO, this condition is not sufficient to validate whether it's cartesian join in join reordering since some filter conditions are pushed to the underlying `DataSource`s.